### ISSUE TITLE
Add option to ignore templating tags

### ISSUE THREAD

**Issue by maxnordlund** [htacg/tidy-html5#691]
After reading #310 I think an option to ignore templating tags would solve it along with any other templating language the user might have.

Taking the [Jinja](http://jinja.pocoo.org/)/[Django](https://docs.djangoproject.com/en/2.0/ref/templates/language/) example from that issue:

```jinja
{% extends "layout.html" %}
{% block body %}
  <ul>
  {% for user in users %}
    <li><a href="{{ user.url }}">{{ user.username }}</a></li>
  {% endfor %}
  </ul>
{% endblock %}
```

If tidy could be, in this case, convinced to ignore `{% .. %}` and `{ ... }` tags then it would parse correctly. Same goes for the popular [mustaches](https://handlebarsjs.com/) [syntax](https://code.angularjs.org/1.6.9/docs/guide/interpolation) `{{ ... }}` and  `<% ... %>` for [ERB](https://ruby-doc.org/stdlib-2.5.0/libdoc/erb/rdoc/ERB.html)/EEX](https://hexdocs.pm/eex/EEx.html).

This is a more general solution, and is similar to the custom tags option that already exists. What do you think?

**Comment by geoffmcl** [comment]
@maxnordlund this seems a repeat of **closed** #310, Nov, 2015, but maybe it is time to look again...

First `tidy` is a **HTML** linter, fixer, tidier, and **not** a `template` tool! It should **only** parse, fix, format `[x]html`...

Having said that, `tidy` does already handle things like `<?`, `<%`, `<#`, and **more**, ..., more or less as transparent text, so maybe there **is** a case for `others` to be handled `better`... perhaps under a **new** suitable option...

It has to be repeated with a config of `--show-body-only yes --fix-uri no` tidy already does quite well with the above sample input, added as [in_691.html](https://github.com/geoffmcl/tidy-test/blob/master/test/input5/in_691.html) to my test repo...

All I can say at this point is that comments, patches, PR's would certainly be reviewed, and provided there were no adverse consequences to tidy's **html** parsing, considered for inclusion...

Look forward to further feedback... thanks...

**Comment by maxnordlund** [comment]
Not quite, it's about the general case and as such I do believe it should be in tidy. You're right that tidy mostly ignores templating languages, but there are places where it doesn't which generates false positives.

Those are when free text is not allowed, like `table > tr > td` and `ul > li` relationships. Unfortunately those are also the ones that commonly have loops, which means they generate a lot of noise when checked with tidy.

I agree that you should need the `--show-body-only yes` option, but URI that contain templating tags should be ignored as you can only know at runtime if the resulting URI is valid or not. For stuff like: `<a href="{{ url_for 'index' }}">Home</a>`.

In order to do this in a sane way, I suggest adding some sort of config for templating start/end tags that tidy matches, and completely ignores, `--templating-tag-start "{" --templating-tag-end "}"` maybe.
